### ISSUE TITLE
Install problems

### ISSUE THREAD

**Issue by patfon1** [StrongKey/fido2#21]
**Describe the bug**
Getting a mssql error when running the install script on ubuntu-18.04.3.

**To Reproduce**
1. login to ubuntu as a normal user and start a terminal
2. download software
3. unzip and run install script; sudo ./install-skfs.sh

**Expected behavior**
Server and database is installed and started. 

**Desktop (please complete the following information):**
 - OS: ubuntu-18.04.3

**Additional context**
user@myserver: $ sudo ./install-skfs.sh 
Installing required linux packages (openjdk, unzip, libaio, ncurses-compat-libs[only applicable for Amazon Linux], rng-tools, curl) ... The installer will skip packages that do not apply or are already installed. HRNGDEVICE=/dev/urandom
Successful
Installing StrongKey FIDO2 Server (SKFS) ... Successful
Installing MariaDB... Successful
mysql binary does not exist or cannot be executed.
user@myserver: $

**Comment by push2085** [comment]
Hi @patfon1 ,
Looks like there is a bug in the install script that is causing this problem. We will address it in the next update. Meanwhile, this is what you can do to proceed. 
1) Install vim or any other editiing tool on the vm

2) Edit the install script
shell> vim install-skfs.sh

3) Remove the following block of code (lines 309-316)

 **_MYSQL_CMD=$(which yum  2>/dev/null)
 
 if [[ ! -z $MYSQL_CMD ]]; then
         :
 else
         echo "mysql binary does not exist or cannot be executed."
         exit 1
fi_** 


4) Run clean up script
shell> sudo ./cleanup.sh

5) Run the install script again
shell> sudo ./install-skfs.sh

**Comment by patfon1** [comment]
Ok, tanks. The install-script continued. 

But, I got another problem during installation; It seems like a log file is missing, the install script is waiting for the file to be created. 

Installing required linux packages (openjdk, unzip, libaio, ncurses-compat-libs[only applicable for Amazon Linux], rng-tools, curl) ... The installer will skip packages that do not apply or are already installed. HRNGDEVICE=/dev/urandom
System has not been booted with systemd as init system (PID 1). Can't operate.
Successful
Installing StrongKey FIDO2 Server (SKFS) ... Successful
Installing MariaDB... Successful
Installing Payara... Successful
Installing JDBC Driver... Successful
Installing OpenDJ... Successful
$Stopping OpenDJ 3.0.0 Directory Service:
$Starting OpenDJ 3.0.0 Directory Service:
touch: cannot touch '/var/lock/subsys/OpenDJ': No such file or directory
Importing default users... Successful
Creating 10M SKFS Internal Database...grep: /usr/local/strongkey/mariadb-10.2.13/log/mysqld-error.log: No such file or directory
.grep: /usr/local/strongkey/mariadb-10.2.13/log/mysqld-error.log: No such file or directory
.grep: /usr/local/strongkey/mariadb-10.2.13/log/mysqld-error.log: No such file or directory
.grep: /usr/local/strongkey/mariadb-10.2.13/log/mysqld-error.log: No such file or directory
.grep: /usr/local/strongkey/mariadb-10.2.13/log/mysqld-error.log: No such file or directory
.grep: /usr/local/strongkey/mariadb-10.2.13/log/mysqld-error.log: No such file or directory
.grep: /usr/local/strongkey/mariadb-10.2.13/log/mysqld-error.log: No such file or directory
.grep: /usr/local/strongkey/mariadb-10.2.13/log/mysqld-error.log: No such file or directory
.^C

I removed that loop got past it and got another problem; 
......
Creating 10M SKFS Internal Database...grep: /usr/local/strongkey/mariadb-10.2.13/log/mysqld-error.log: No such file or directory
done
ERROR 2002 (HY000): Can't connect to local MySQL server through socket '/usr/local/strongkey/mariadb-10.2.13/log/mysqld.sock' (2)
ERROR 2002 (HY000): Can't connect to local MySQL server through socket '/usr/local/strongkey/mariadb-10.2.13/log/mysqld.sock' (2)
ERROR 2002 (HY000): Can't connect to local MySQL server through socket '/usr/local/strongkey/mariadb-10.2.13/log/mysqld.sock' (2)
ERROR 2002 (HY000): Can't connect to local MySQL server through socket '/usr/local/strongkey/mariadb-10.2.13/log/mysqld.sock' (2)
ERROR 2002 (HY000): Can't connect to local MySQL server through socket '/usr/local/strongkey/mariadb-10.2.13/log/mysqld.sock' (2)
ERROR 2002 (HY000): Can't connect to local MySQL server through socket '/usr/local/strongkey/mariadb-10.2.13/log/mysqld.sock' (2)
$Starting Glassfish 4 Application Server:
touch: cannot touch '/var/lock/subsys/glassfishd': No such file or directory
Exception in thread "main" java.lang.NoClassDefFoundError: sun/misc/BASE64Encoder
        at java.base/java.lang.ClassLoader.defineClass1(Native Method)
        at java.base/java.lang.ClassLoader.defineClass(ClassLoader.java:1016)
        at java.base/java.security.SecureClassLoader.defineClass(SecureClassLoader.java:174)
        at java.base/jdk.internal.loader.BuiltinClassLoader.defineClass(BuiltinClassLoader.java:802)
        at java.base/jdk.internal.loader.BuiltinClassLoader.findClassOnClassPathOrNull(BuiltinClassLoader.java:700)
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClassOrNull(BuiltinClassLoader.java:623)
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:581)
        at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:521)
        at com.sun.appserv.management.client.prefs.MemoryHashLoginInfoStore.<clinit>(MemoryHashLoginInfoStore.java:63)
        at com.sun.appserv.management.client.prefs.LoginInfoStoreFactory.getDefaultStore(LoginInfoStoreFactory.java:77)
        at com.sun.enterprise.admin.cli.remote.RemoteCLICommand.initializeAuth(RemoteCLICommand.java:981)
        at com.sun.enterprise.admin.cli.remote.RemoteCLICommand.prepare(RemoteCLICommand.java:633)
        at com.sun.enterprise.admin.cli.CLICommand.execute(CLICommand.java:300)
        at com.sun.enterprise.admin.cli.remote.RemoteCLICommand.execute(RemoteCLICommand.java:612)
        at com.sun.enterprise.admin.cli.AdminMain.executeCommand(AdminMain.java:371)
        at com.sun.enterprise.admin.cli.AdminMain.doMain(AdminMain.java:305)
        at org.glassfish.admin.cli.AsadminMain.main(AsadminMain.java:56)
Caused by: java.lang.ClassNotFoundException: sun.misc.BASE64Encoder
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:583)
        at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:521)
        ... 18 more
Exception in thread "main" java.lang.NoClassDefFoundError: sun/misc/BASE64Encoder

**Comment by push2085** [comment]
Hi @patfon1 ,
Do not remove anything else from the script, that will cause more problems. 
Is this VM on a cloud or internal?
do you have internet access on this vm? The script tries to install packages and download things from the internet and If it doesnt get something it can mess up the mysql install.

Make sure you have enough space on the disk as well to create all the necessary files.

**Comment by push2085** [comment]
I also see thsi one message in there which is unuslay
"System has not been booted with systemd as init system (PID 1). Can't operate."

Can you run clean up again, reboot the vm and then make sure it has the space and internet access before running the install script again.

**Comment by patfon1** [comment]
I'm running ubuntu on wmvare and I have internet access.

The install scripts stops in a loop and is waiting for a log file to be created (mysqld-error.log). But it does not get created for some reason. 

This is my output; 
Installing required linux packages (openjdk, unzip, libaio, ncurses-compat-libs[only applicable for Amazon Linux], rng-tools, curl) ... The installer will skip packages that do not apply or are already installed. HRNGDEVICE=/dev/urandom
Successful
Installing StrongKey FIDO2 Server (SKFS) ... Successful
Installing MariaDB... Successful
Installing Payara... Successful
Installing JDBC Driver... Successful
Installing OpenDJ... Successful
Importing default users... Successful
Creating 10M SKFS Internal Database...grep: /usr/local/strongkey/mariadb-10.2.13/log/mysqld-error.log: Fila eller mappa finnes ikke
.grep: /usr/local/strongkey/mariadb-10.2.13/log/mysqld-error.log: Fila eller mappa finnes ikke
.grep: /usr/local/strongkey/mariadb-10.2.13/log/mysqld-error.log: Fila eller mappa finnes ikke
.grep: /usr/local/strongkey/mariadb-10.2.13/log/mysqld-error.log: Fila eller mappa finnes ikke
.grep: /usr/local/strongkey/mariadb-10.2.13/log/mysqld-error.log: Fila eller mappa finnes ikke
.grep: /usr/local/strongkey/mariadb-10.2.13/log/mysqld-error.log: Fila eller mappa finnes ikke
.grep: /usr/local/strongkey/mariadb-10.2.13/log/mysqld-error.log: Fila eller mappa finnes ikke
.grep: /usr/local/strongkey/mariadb-10.2.13/log/mysqld-error.log: Fila eller mappa finnes ikke
.grep: /usr/local/strongkey/mariadb-10.2.13/log/mysqld-error.log: Fila eller mappa finnes ikke
.grep: /usr/local/strongkey/mariadb-10.2.13/log/mysqld-error.log: Fila eller mappa finnes ikke
.grep: /usr/local/strongkey/mariadb-10.2.13/log/mysqld-error.log: Fila eller mappa finnes ikke
.^C

**Comment by push2085** [comment]
Hi @patfon1 ,
I am not able to reproduce this issue. I tried deploying the same ubuntu version in GCP and went through the install. The only thing I had to change was the remove the block of code as mentioned above.

Let me know If there is anything specific I can do to try to reproduce this. 
Another thing I would suggest is to re create the VM instead of using the existing one and try one more time.
